python/test/unit/hopper/test_gemm.py

```diff
@@ -264,11 +266,11 @@ def matmul_kernel(
                              for use_tma_store in [False, True]
                              for trans_a in [False, True]
                              for trans_b in [False, True]
-                             for trans_c in [False, True]
+                             for trans_output in [False, True]
                              for num_stages in [3]
                              for enable_ws in [False, True]
-                             if not (epilogue == 'chain-dot' and (shape_w_c[5] is not None or shape_w_c[0] != shape_w_c[1]))
-                         ] + [(*shape_w_c, trans_a, trans_b, trans_c, 'none', out_dtype, use_tma_store, num_stages, enable_ws)
+                             if not (epilogue == 'chain-dot' and (shape_w_c[6] is not None or shape_w_c[1] != shape_w_c[6]))
```

Comment:
`not (shape_w_c[6] is not None or shape_w_c[1] != shape_w_c[6])` and `shape_w_c[6] is None and shape_w_c[1] == shape_w_c[6]` are equivalent. Actually we need `shape_w_c[6] is None or shape_w_c[1] == shape_w_c[6]` here. So it is more reasonable to change this condition to `if not (epilogue == 'chain-dot' and (shape_w_c[6] is not None and shape_w_c[1] != shape_w_c[6]))`.

Reply:
hmm i think you are right and nice catch.
This actually mistakenly enlarge the scope that needs to be skipped. 
I'll do some test locally first and should append a PR later. Thanks.